src/map/skill.c

```diff
@@ -7306,27 +7306,37 @@ static int skill_castend_nodamage_id(struct block_list *src, struct block_list *
 			break;
 
 		case PR_STRECOVERY:
-			if(status->isimmune(bl)) {
-				clif->skill_nodamage(src,bl,skill_id,skill_lv,0);
+			if (status->isimmune(bl) != 0) {
+				clif->skill_nodamage(src, bl, skill_id, skill_lv, 0);
 				break;
 			}
-			if (tsc && tsc->opt1) {
-				status_change_end(bl, SC_FREEZE, INVALID_TIMER);
-				status_change_end(bl, SC_STONE, INVALID_TIMER);
-				status_change_end(bl, SC_SLEEP, INVALID_TIMER);
-				status_change_end(bl, SC_STUN, INVALID_TIMER);
-				status_change_end(bl, SC_WHITEIMPRISON, INVALID_TIMER);
+
+			if (bl->type == BL_PC && tstatus->def_ele != ELE_UNDEAD) {
```

Comment:
you just broke the whole thing.
1. you should only check undead using battle->check_undead otherwise it ignores server configuration
2. by adding the bl->type == BL_PC you broke the skill behavior, it should be able to status recovery non-pc but not on undead property objects
3. the skill shall fail on undead property players, you should instead add a check for PC object their element and do `clif->skill_nodamage(src, bl, skill_id, skill_lv, 0);` similar to the isimmune check above. (or simply merge it there), while on non-PC objects that are undead it shall only apply the blind formula down there.

Reply:
> 1. you should only check undead using battle->check_undead otherwise it ignores server configuration

Yeah, my fault....

> 2. by adding the bl->type == BL_PC you broke the skill behavior, it should be able to status recovery non-pc but not on undead property objects

Does it cure status effects on Mercenaries/Homunculi? I just tested monsters and they are not cured, so I assumed it's only for players. In any case, monsters should be excluded.

Reply:
The bug is reporting in pre-re and you're testing on renewal.

Reply:
I'm curently unable to test, neither on pre-RE, nor on RE. Would you be so nice and give me some details, since you seem to know the correct mechanism for both?

Reply:
Removed the BL_PC restriction and replaced `tstatus->def_ele != ELE_UNDEAD` with a call of `battle_check_undead()`.